### ISSUE TITLE
build fail on release 1.65 and 1.66

### ISSUE THREAD

**Issue by tristanRW** [canonical/snapd-glib#175]
When fetching the newest release 1.65 and running `meson setup -Dprefix=/usr _builddir_` followed by `ninja -C _builddir_` ninja fails at [147/205]: tests/tests-markdown-qt6.

I'm at my wits end with this, after researching and trying for days, but can provide any logs needed to someone skilled enough.

[meson.log](https://github.com/user-attachments/files/16956424/meson.log)
[ninja.log](https://github.com/user-attachments/files/16956425/ninja.log)

**Comment by sergio-costas** [comment]
Try to delete the _builddir_ folder before running setup. The log says that it's already configured, so it maybe is using an old setup.

**Comment by tristanRW** [comment]
Please refer to the ninja.log. The builddir was configured using the command given. I will attach the correct meson.log soon.

**Comment by sergio-costas** [comment]
The point is that I need info about which system type and architecture is this being compiled on, and that is shown in the meson setup log...

**Comment by tristanRW** [comment]
I finally updated both logs. Should contain all info now. Sorry for the delay and thank you for looking into it so quickly.

**Comment by sergio-costas** [comment]
Mmm... Under which Ubuntu version are you building this?

**Comment by sergio-costas** [comment]
OK, I tested it under Ubuntu Oracular, which has the same gcc version, and still builds... Can you tell me under which system are you building it, please?

**Comment by tristanRW** [comment]
Hi, I am building on Arch x86_64, qt5 and qt6 are installed for plasma6-desktop.
Where should I check for the .so files (where should they be)?

**Comment by sergio-costas** [comment]
Check where in your system are these files:

```
libQt6Qml.so
libQt6QmlBuiltins.a
libQt6Network.so
libQt6Core.so
```
If they are at /usr/lib, then check with `ldd` if all the dependencies are in your system.

**Comment by tristanRW** [comment]
Sorry, I am new to this. Which file should I check using ldd ?

**Comment by sergio-costas** [comment]
Each one of those four libraries, wherever they are. Let's say that they are at /usr/lib/x86_64-linux-gnu; then you should do

`ldd /usr/lib/x86_64-linux-gnu/libQt6Qml.so`

and check that all entries point to another file.

Although, if KDE works, that should not be a problem...

**Comment by tristanRW** [comment]
Ok, so:

- Permissions on libQt6QmlBuiltins.a look funky, since root may no execute it, but I don't know if thats intended `.rw-r--r--  148k root 2024-06-18 23:15 libQt6QmlBuiltins.a`. In any case I cannot run ldd on it.
- I have checked the other .so files using ldd. [ldd.log](https://github.com/user-attachments/files/17014960/ldd.log) and it seems ok. At least no errors.

**Comment by tristanRW** [comment]
Any news on this issue? It persists in v1.66 sadly.
Appended the build-logs for version 1.66. Seems to be the same problem still.

[meson.log](https://github.com/user-attachments/files/17227665/meson.log)
[ninja.log](https://github.com/user-attachments/files/17227666/ninja.log)

**Comment by sergio-costas** [comment]
I see that you are still using v1.66 instead of `master`. Can you try with it, instead, please?

**Comment by sergio-costas** [comment]
Ok, I installed an Arch system in a docker container, and I'm having trouble too. I'll try to fix it. Give me a moment...

**Comment by sergio-costas** [comment]
Ok, I did everything in my hand, but I'm unable to find why it can't find the symbols. Can you post this in an Arch forum and ask there if someone can check it, please?

**Comment by aleixpol** [comment]
Despite the issue getting closed (???) the project still cannot be built.

**Comment by sergio-costas** [comment]
But cannot be built in Arch, or in general? Does `master` fail too?

**Comment by aleixpol** [comment]
I am using ArchLinux so the distinction doesn't make a big difference to me.

Yes, I've been trying master. It fails both for gcc and clang.

**Comment by sergio-costas** [comment]
Ok, I've been using an Arch docker, and there are some odd things. Basically it can't link anything with the compiled library. I'm not sure why, but there is something odd when using readelf: in the libQt6Core.so file, it shows these two lines:

  0c4:  2d (GCC_3.0)      2e (GLIBC_2.33)   12 (GLIBC_2.2.5)  18 (CXXABI_1.3.11)
  17c:  37 (GCC_3.3.1)    17 (GLIBC_2.7)    12 (GLIBC_2.2.5)  38 (GLIBC_2.25) 

(with that CXXABI), while in the ubuntu library it shows

  0c8:  18 (GLIBC_2.34)   2b (GCC_3.0)      2c (GLIBC_2.33)    1 (*global*)   
  17c:  1a (GLIBC_2.3.4)  34 (GCC_3.3.1)    14 (GLIBC_2.7)     f (GLIBC_2.2.5)

(with *global*). Meanwhile, the compiled libsnapd-qt-2.so.1 shows

  120:   a (GCC_3.0)       1 (*global*)      1 (*global*)      1 (*global*)

in the ubuntu version, and 

  120:   1 (*global*)      1 (*global*)      1 (*global*)      a (GCC_3.0)

in the arch version... Maybe the problem is there? An ABI problem?

**Comment by sergio-costas** [comment]
@jhenstridge I've been doing all kind of tests, but nothing. Any idea why is this happening? I'm afraid that it could be a problem with Qt6.8.

**Comment by sergio-costas** [comment]
Ok, it builds fine in Fedora with Qt 6.8.2, so clearly this is some problem in Arch, I'm afraid...

**Comment by sergio-costas** [comment]
The problem itself is when trying to *link* other code with the generated library: it can't find any of the exported symbols. But why? I don't know.

**Comment by sergio-costas** [comment]
Mmm... But why should it "overflow" to our own library?

**Comment by aleasto** [comment]
Because it writes the `QT_USE_PROTECTED_VISIBILITY` config to `/usr/include/qt6/QtCore/qconfig.h` which thus affects all Qt libraries

**Comment by sergio-costas** [comment]
Oh! That makes sense. Thanks!

**Comment by sergio-costas** [comment]
Closing then, since it's not our fault. Thanks for the help, @aleasto !

**Comment by aleasto** [comment]
Yeah that makes sense. But is that really how we're supposed to deal with `QT_USE_PROTECTED_VISIBILITY`?

**Comment by sergio-costas** [comment]
That's what CALAMARES did... https://github.com/calamares/calamares/commit/4eba859236777c5764ebd3db883a303ed810ee7b
